lua/customroles/werewolf.lua

```diff
@@ -0,0 +1,797 @@
+WEREWOLF_NIGHT_ALWAYS_SHOW = 3
+
+WEREWOLF_TIMER_NONE = 0
+WEREWOLF_TIMER_WEREWOLVES = 1
+WEREWOLF_TIMER_ALL = 2
+
+WEREWOLF_FOG_NONE = 0
+WEREWOLF_FOG_NONWEREWOLVES = 1
+WEREWOLF_FOG_ALL = 2
+
+WEREWOLF_VISION_NEVER = 0
+WEREWOLF_VISION_TRANSFORMED = 1
+WEREWOLF_VISION_ALWAYS = 2
+
+WEREWOLF = {
+    isNight = false,
+    nightTime = 0
+}
+
+if SERVER then
+    AddCSLuaFile()
+
+    util.AddNetworkString("TTT_WerewolfSetNight")
+
+    ------------------------
+    -- DAY/NIGHT TRACKING --
+    ------------------------
+
+    local oldPlayerModels = {}
+
+    WEREWOLF.ChangeTime = function(forceDay, hideMessages, blockTimers)
+        local min, max
+        if WEREWOLF.isNight or forceDay then
+            min = werewolf_day_length_min:GetInt()
+            max = werewolf_day_length_max:GetInt()
+            WEREWOLF.isNight = false
+        else
+            min = werewolf_night_length_min:GetInt()
+            max = werewolf_night_length_max:GetInt()
+            WEREWOLF.isNight = true
+        end
+
+        if max < min then
+            max = min
+        end
+        local length = math.random(min, max)
+        if blockTimers then
+            length = 0
+            WEREWOLF.nightTime = 0
+        else
+            WEREWOLF.nightTime = CurTime() + length
+        end
+
+        net.Start("TTT_WerewolfSetNight")
+        net.WriteBool(WEREWOLF.isNight)
+        net.WriteUInt(length, 10)
+        net.Broadcast()
+
+
+        local night_visibility_mode = werewolf_night_visibility_mode:GetInt()
+        for _, v in player.Iterator() do
+            if not hideMessages then
+                if v:IsActiveWerewolf() or night_visibility_mode >= WEREWOLF_NIGHT_SHOW_IF_HAS_WEREWOLF then
+                    if WEREWOLF.isNight then
+                        v:QueueMessage(MSG_PRINTTALK, "Night falls...")
+                    else
+                        v:QueueMessage(MSG_PRINTTALK, "The sun rises...")
+                    end
+                end
+            end
+
+            if v:IsActiveWerewolf() then
+                if WEREWOLF.isNight then
+                    local drop_weapons = werewolf_drop_weapons:GetBool()
+                    if drop_weapons then
+                        for _, wep in pairs(v:GetWeapons()) do
+                            local class = WEPS.GetClass(wep)
+                            if class ~= "weapon_zm_improvised" and class ~= "weapon_wwf_claws" and wep.AllowDrop then
+                                v:DropWeapon(wep)
+                            end
+                        end
+                    end
+                    v:Give("weapon_wwf_claws")
+                    v:SelectWeapon("weapon_wwf_claws")
+
+                    local transform_model = werewolf_transform_model:GetBool()
+                    if transform_model then
+                        oldPlayerModels[v:SteamID64()] = v:GetModel()
+                        SetMDL(v, "models/player/stenli/lycan_werewolf.mdl")
```

Comment:
Add ConVar for this and check that it exists before trying to use it.
Might be able to simplify it by using a convar change callback and disabling `ttt_werewolf_transform_model` if the set model doesn't exist. Though you may have to handle the initial load of the convars separately because I'm not sure if the callback is called when the convars are first created and the values are the defaults.